### ISSUE TITLE
Learning rate estimation questions in 2.0.0rc2 

### ISSUE THREAD

**Issue by qwe795138426** [LoganBooker/prodigy-plus-schedule-free#31]
When I tried version 2.0.0, I tried the recommended speed and found two questions:
1. I felt that the learning rate scaling was not as good as training without speed, and the adjustment seemed more difficult; (it was reflected in the fact that the learning rate would be adjusted to a similar value after adjusting the learning rate)
2. TE occasionally increased abnormally, and I could not find out which setting conflicted with it;(It is normal in some cases, but very stubborn when it is abnormal.)
Here is a problematic picture, the learning rate is set to TE:0.25, Unet:1, I hope it can help you judge it.
![Image](https://github.com/user-attachments/assets/43762adb-d3e2-40dd-9427-04c3aea6f1d8)

![Image](https://github.com/user-attachments/assets/21520827-8e76-4ddd-9c45-0b2f5606a4b8)

**Comment by qwe795138426** [comment]
I got it. It's due to max_grad_norm = 1

**Comment by qwe795138426** [comment]
I'm back again.

After removing the interference factors, I continued to try and found that TE was still difficult to control (I tried various weight coefficients from 1/10 to 1/3). I found that the problem of abnormal rise was indeed alleviated, but it was still difficult to control the ratio of TE/Unet. I felt that there was something that caused TE to overestimate.
(Under SPEED, if the weight coefficient of TE is less than 0.5, it seems that retaliatory overestimation may occur)

Then I tried to remove SPEED and found that when the coefficients were equal to 1 at the same time, TE would continue to increase during training, and when the te/ue ratio was less than 1, UNET would overestimate and stabilize at a higher value around 500-800 steps. The final LR ratio was very sensitive to the TE/UE ratio.

I wonder what may have caused this situation.

The following are images for reference, I hope it will be helpful,

![Image](https://github.com/user-attachments/assets/3710eb52-3c49-4600-8376-bb1c32d93340)
（optimizer_args = [ "eps=None", "weight_decay=0.001", "d0=1e-5" ]，
 learning_rate = 1
unet_lr = 1.2
text_encoder_lr = 1
）
